src/commands/set.js

```diff
@@ -41,24 +43,34 @@ const writeConfigToFile = newConfig => {
 
 const checkBoolean = value => ['true', 'false'].includes(value);
 
-const setNestedConfigProperty = newValue => (obj, pathComponent, i, path) => {
-	if (i === path.length - 1) {
+const setNestedConfigProperty = newValue => (
+	obj,
+	pathComponent,
+	i,
+	dotNotationArray,
+) => {
+	if (i === dotNotationArray.length - 1) {
+		if (obj[pathComponent] === undefined) {
+			throw new ValidationError(
+				`Config file could not be written: property ${pathComponent} was not found. It looks like your configuration file is corrupted. Please check the file or run 'lisky clean' to remove it (a fresh default configuration file will be created when you run Lisky again.`,
+			);
+		}
 		// eslint-disable-next-line no-param-reassign
 		obj[pathComponent] = newValue;
 		return config;
 	}
 	return obj[pathComponent];
```

Comment:
If `obj` is undefined, shouldn't we also throw the above `ValidationError`?

Reply:
`obj` undefined should be checked above.